### PR TITLE
Add deployment related fields to FeedSource

### PR DESCRIPTION
**Pull request by br648**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR enhances `FeedSource` to include `deployedFeedVersionId`, `deployedFeedVersionStartDate` and `deployedFeedVersionEndDate`. These are derived from the feed version used in the latest deployment.

### CODE REVIEW

File: src/main/java/com/conveyal/datatools/manager/models/FeedSource.java [hunk 1]
```diff
@@ -461,6 +462,14 @@ public String latestVersionId() {
         return latest != null ? latest.id : null;
     }
 
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @JsonView(JsonViews.UserInterface.class)
+    @JsonProperty("latestVersionEndDate")
+    public LocalDate latestVersionEndDate() {
+        FeedVersion latest = retrieveLatest();
```
Comment (miles-grant-ibigroup) [inline src/main/java/com/conveyal/datatools/manager/models/FeedSource.java]: Does `retrieveLatest` retrieve the latest **deployed** version? If so, would it be possible to also expose the id of this version?

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Glad the change is this easy! Hopefully... 

Just one piece of info missing and then we should be good to go!

File: src/main/java/com/conveyal/datatools/manager/models/FeedSource.java (hunk 1 above)
Comment (br648) [reply src/main/java/com/conveyal/datatools/manager/models/FeedSource.java]: If I have understood correctly, please see latest update.

**Review by miles-grant-ibigroup — APPROVED** [review]
Works great and code looks great too very java-ey oh my that Java 11 upgrade is paying off

File: src/main/java/com/conveyal/datatools/manager/models/FeedSource.java [hunk 2]
```diff
@@ -461,6 +462,61 @@ public String latestVersionId() {
...
+
+    /**
+     * Find the latest deployment containing a feed version for a feed source.
+     */
+    private FeedVersion getDeployedFeedVersion() {
+        if (deployedFeedVersion != null) {
+            // If the deployed feed version is already defined use this instead of recreating.
+            return deployedFeedVersion;
+        }
+        Collection<Deployment> deployments = Persistence.deployments.getFiltered(
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/datatools/manager/models/FeedSource.java]: Nitpick: I would have done
```
if (deployedFeedVersion == null ) {
    Collection<Deployment> deployments = ...;
    ...
}
return deployedFeedVersion;
```

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM. Let's wait until the UI portion is approved before merging.

**Comment by miles-grant-ibigroup** [comment]
Good call @binh-dam-ibigroup. We actually need `deployedFeedVersionStartDate` as well. To handle the case where a deployed feed version starts in the future but the latest published version starts in the past!

File: src/main/java/com/conveyal/datatools/manager/models/FeedSource.java (hunk 2 above)
Comment (br648) [reply src/main/java/com/conveyal/datatools/manager/models/FeedSource.java]: Updated. Thanks.

**Comment by miles-grant-ibigroup** [comment]
Yep it works great thanks so much!

**Pull request merged**